java/java-impl/src/com/intellij/codeInsight/completion/JavaGenerateMemberCompletionContributor.java

```diff
@@ -45,38 +46,84 @@ public static void fillCompletionVariants(CompletionParameters parameters, Compl
     PsiElement position = parameters.getPosition();
     if (psiElement(PsiIdentifier.class).withParents(PsiJavaCodeReferenceElement.class, PsiTypeElement.class, PsiClass.class).
       andNot(JavaKeywordCompletion.AFTER_DOT).accepts(position)) {
+      int priority = -1;
+      boolean followedByWhitespace = psiElement()
+        .withSuperParent(
+          2, // Go to PsiTypeElement
+          psiElement().beforeSiblingSkipping(
+            psiElement(PsiErrorElement.class), // If no annotations are present, the Psi will contain a "identifier expected" node right after
+            psiElement(PsiWhiteSpace.class)
+              //.withText(StandardPatterns.string().matches("(?s) *\n *\n.*")) // Require line to be followed by empty line
+          )).accepts(position);
+      if (followedByWhitespace) {
+        priority = 50;
+      }
       PsiElement prevLeaf = PsiTreeUtil.prevVisibleLeaf(position);
       PsiModifierList modifierList = PsiTreeUtil.getParentOfType(prevLeaf, PsiModifierList.class);
+
+      boolean hasOverrideAnnotation = false;
       if (modifierList != null) {
         String fileText = position.getContainingFile().getText();
+        String prefix = fileText.substring(modifierList.getTextRange().getStartOffset(), parameters.getOffset());
+        PsiAnnotation[] annotations = PsiTreeUtil.getChildrenOfType(modifierList, PsiAnnotation.class);
+        if (annotations != null) {
+          for (PsiAnnotation annotation : annotations) {
+            if (Override.class.getName().equals(annotation.getQualifiedName())) {
+              hasOverrideAnnotation = true;
```

Comment:
PsiModifierList.findAnnotation?

Reply:
Cool, I didn't know that existed. I take it `PsiModifierList.hasAnnotation` is an even better choice here?

Reply:
True!